### DISCUSSION TITLE
Using forme with sequel_polymorphic

### DISCUSSION THREAD

**Discussion by okliv** [jeremyevans/forme#100] [General]
i want to implement select dropdown with [polymorphic](https://github.com/jackdempsey/sequel_polymorphic#schema) field:

```ruby
f.input :attachable, type: :select, optgroups: {Notes: Note, Posts: Post}, value_method: :id, text_method: :name, value: f.obj.attachable
```

(how) can i do it using one dropdown?

thanks

(my real case has more sense but less clear)

**Comment by jeremyevans** [comment]
I don't have any experience with sequel-polymorphic, but basic use of optgroups would be:

```ruby
 Forme::Form.new.input(:select, :optgroups=>{Notes: [['a', 1], ['b', 2]], Posts: [['c', 1], ['d', 3]]})
# <select><optgroup label="Notes"><option value="1">a</option><option value="2">b</option></optgroup>
# <optgroup label="Posts"><option value="1">c</option><option value="3">d</option></optgroup></select>
```

I think the main problem you will have with this approach is you don't know when the form is submitted which type is submitted.  You could do something like:

```
Forme::Form.new.input(:select,
  :optgroups=>{Notes: [['a', 'note-1'], ['b', 'note-2']], Posts: [['c', 'post-1'], ['d', 'post-3']]})
```

And parse the attachable type information out of the submitted value.  That's kind of ugly, but should allow a single drop down to work.

For this type of thing, selecting the attachable type first, then filling in the attachable options for that type is probably a better approach.  That requires some Javascript, and it appears you want to avoid that approach.

In general, I advise against using polymorphic associations.

**Reply by okliv** [reply]
Yeah, two dropdowns are over-complication for my form but i agree that this is better approach in case if there is no pre-designed way to manage polymorphic type somehow in hidden field, for example (next i will try is to do it with js) Thanks a lot!
